tools/walletextension/accountmanager/account_manager.go

```diff
@@ -252,38 +252,55 @@ func executeSubscribe(client *rpc.EncRPCClient, req *RPCRequest, _ *interface{},
 }
 
 func executeCall(client *rpc.EncRPCClient, req *RPCRequest, resp *interface{}) error {
-	if req.Method == rpc.RPCCall || req.Method == rpc.RPCEstimateGas {
-		// RPCCall is a sensitive method that requires a viewing key lookup but the 'from' field is not mandatory in geth
-		//	and is often not included from metamask etc. So we ensure it is populated here.
+	// never modify the original request as it might be reused
+	clonedRequest := RPCRequest{
```

Comment:
Maybe worth giving the `RPCRequest` struct a `Clone()` method? Probably quite a common thing to do for requests and makes this a bit more succinct.

Reply:
Good idea 👍